src/transformers/generation/utils.py

```diff
@@ -4591,11 +4591,10 @@ def assisted_decoding(
             cur_len = input_ids.shape[-1]
 
             #  1. Fetch candidate sequences from a `CandidateGenerator`
-            candidate_input_ids, candidate_logits = candidate_generator.get_candidates(
-                input_ids.to(candidate_generator.assistant_model.device)
-            )
+            candidate_input_ids, candidate_logits = candidate_generator.get_candidates(input_ids)
```

Comment:
in a nutshell, `input_ids` should only be moved on some `candidate_generator` classes -- the move was delegated to the relevant classes